internal/internal_workflow_test.go

```diff
@@ -1417,6 +1418,10 @@ func (uc *updateCallback) Reject(err error) {
 }
 
 func (uc *updateCallback) Complete(success interface{}, err error) {
+	// cache update result so we can dedup duplicate update IDs
+	if uc.env.impl.updateMap != nil {
+		uc.env.impl.updateMap[uc.env.impl.currentUpdateId] = success
```

Comment:
The error would also need to be cached if we want to go with the caching approach

Reply:
the error here represents the users error they returned from the update handler